x-pack/plugins/ml/public/application/jobs/new_job/job_from_lens/create_job.ts

```diff
@@ -0,0 +1,284 @@
+  }
+}
+
+export async function canCreateADJob(
+  vis: LensSavedObjectAttributes,
+  query: Query | undefined,
+  filters: Filter[] | undefined,
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  try {
+    const jobItems = await createADJobFromLensSavedObject(
+      vis,
+      query ?? getDefaultQuery(),
+      filters ?? [],
+      dataViewClient,
+      kibanaConfig
+    );
+
+    return !!jobItems;
+  } catch (error) {
+    return false;
+  }
+}
+
+async function createADJobFromLensSavedObject(
+  vis: LensSavedObjectAttributes,
+  query: Query,
+  filters: Filter[],
+  dataViewClient: DataViewsContract,
+  kibanaConfig: IUiSettingsClient
+) {
+  const { fields, timeField, splitField, dataView } = await extractFields(vis, dataViewClient);
+
+  const jobConfig = createEmptyJob();
+  const datafeedConfig = createEmptyDatafeed(dataView.title);
+
+  const mainQuery = query?.query !== '' ? query : vis.state.query;
+
+  const { combinedQuery } = createQueries(
+    { query: mainQuery, filter: filters },
+    dataView,
+    kibanaConfig
+  );
+  datafeedConfig.query = combinedQuery;
+
+  jobConfig.analysis_config.detectors = createDetectors(fields, splitField);
+
+  jobConfig.data_description.time_field = timeField.sourceField;
+  jobConfig.analysis_config.bucket_span = '15m';
+  if (splitField) {
+    jobConfig.analysis_config.influencers = [splitField.sourceField];
+  }
+
+  const createdBy =
+    splitField || jobConfig.analysis_config.detectors.length > 1
+      ? CREATED_BY_LABEL.MULTI_METRIC
+      : CREATED_BY_LABEL.SINGLE_METRIC;
+
+  return {
+    jobConfig,
+    datafeedConfig,
+    createdBy,
+  };
+}
+
+async function extractFields(vis: LensSavedObjectAttributes, dataViewClient: DataViewsContract) {
+  const visualization = vis.state.visualization as { layers: XYDataLayerConfig[] };
+  const indexpattern = vis.state.datasourceStates.indexpattern as IndexPatternPersistedState;
+
+  const compatibleLayers = visualization.layers.filter(
+    ({ layerType, seriesType }) =>
+      layerType === COMPATIBLE_LAYER_TYPE && COMPATIBLE_SERIES_TYPES.includes(seriesType)
+  );
+
+  const [firstCompatibleLayer] = compatibleLayers;
+
+  const compatibleIndexPatternLayer = Object.entries(indexpattern.layers).find(
+    ([id]) => firstCompatibleLayer.layerId === id
+  );
+  if (compatibleIndexPatternLayer === undefined) {
+    throw Error(
+      'Visualization does not contain any layers which can be used for creating an anomaly detection job.'
+    );
+  }
+
+  const [layerId, layer] = compatibleIndexPatternLayer;
+
+  const columns = getColumns(layer);
+  const timeField = Object.values(columns).find(({ dataType }) => dataType === 'date');
+  if (timeField === undefined) {
+    throw Error('Cannot find a date field.');
+  }
+
+  const fields = firstCompatibleLayer.accessors.map((a) => columns[a]);
+
+  const splitField = firstCompatibleLayer.splitAccessor
+    ? columns[firstCompatibleLayer.splitAccessor]
```

Comment:
Doesn't check for the type of the field - if it's a number, job creation fails like this:
<img width="360" alt="Screenshot 2022-05-12 at 13 09 53" src="https://user-images.githubusercontent.com/1508364/168059494-9a13095e-5669-437e-8a6c-e69bfb43289e.png">

Reply:
Fixed in https://github.com/elastic/kibana/pull/129762/commits/3ba9b7cc140748fb6916d50eb94c96f66ef90490